vcr/cassette.py

```diff
@@ -266,6 +266,9 @@ def _responses(self, request):
                 yield index, response
 
     def can_play_response_for(self, request):
+        # Deepcopy is here because mutation of `request` will corrupt the
+        # real request.
+        request = copy.deepcopy(request)
```

Comment:
There is an idea at https://github.com/kevin1024/vcrpy/issues/174#issuecomment-1242432327 to change the `request = copy.copy(request)` in…
https://github.com/kevin1024/vcrpy/blob/1a3bc67c7c9dcede68c1b072fff2676d221b230b/vcr/config.py#L221-L227
…to `deepcopy` instead, and it seems cleaner if also fixing what you're intending to fix. Have you tried? What do you think?

PS: Disclaimer: I only know part of the picture here, this will need review by someone else on top.

Reply:
Are you saying if I changed the `copy` to `deepcopy` in `before_record_request` I could remove the change in `can_play_response`?

Reply:
Well it passes my tests, so I will update with this. Seems a lot cleaner.

Reply:
@abramclark the changes to `can_play_response_for` are still included. Did you forget to drop them? Does the test still pass without?